tests/phpunit/tests/media.php

```diff
@@ -3462,6 +3452,184 @@ public function data_wp_get_attachment_image_decoding_attr() {
 		);
 	}
 
+	/**
+	 * Test whether wp_decoding_enabled() returns the expected value.
+	 *
+	 * @ticket       58892
+	 * @dataProvider data_wp_decoding_enabled_tag_name_defaults
+	 *
+	 * @covers ::wp_decoding_enabled
+	 *
+	 * @param string $tag_name Tag name.
+	 * @param bool   $expected Expected return value.
+	 */
+	public function test_wp_decoding_enabled( $tag_name, $expected ) {
+		if ( $expected ) {
+			$this->assertTrue( wp_decoding_enabled( $tag_name, 'the_content' ) );
+		} else {
+			$this->assertFalse( wp_decoding_enabled( $tag_name, 'the_content' ) );
+		}
+	}
+
+	/**
+	 * Data provider for test_wp_decoding_enabled().
+	 *
+	 * @return array {
+	 *  @type array $test {
+	 *      @type string  $tag_name The tag name.
+	 *      @type bool    $expected Whether tag name is supported.
+	 *  }
+	 * }
+	 */
+	public function data_wp_decoding_enabled_tag_name_defaults() {
+		return array(
+			'img => true'            => array( 'img', true ),
+			'iframe => false'        => array( 'iframe', false ),
+			'arbitrary tag => false' => array( 'blink', false ),
+		);
+	}
+
+	/**
+	 * Tests that the `wp_get_loading_optimization_attributes()` function should add
+	 * the 'decoding' attribute.
+	 *
+	 * @ticket 58892
+	 *
+	 * @dataProvider data_add_decoding_attr
+	 *
+	 * @param string $tag_name The tag name.
+	 * @param string $decoding The value for the 'decoding' attribute. 'no value' for default.
+	 * @param string $expected The expected `img` tag.
+	 */
+	public function test_add_decoding_attr( $tag_name, $decoding, $expected ) {
+			add_filter(
+				'wp_decoding_value',
+				static function( $value ) use ( $decoding ) {
+					return $decoding;
+				}
+			);
+
+		$this->assertEquals(
```

Comment:
```suggestion
		add_filter(
			'wp_decoding_value',
			static function( $value ) use ( $decoding ) {
				return $decoding;
			}
		);

		$this->assertEquals(
```
Correct the indentation